### PR TITLE
Auto: implement issue #199

### LINKED ISSUE #199: [Test] Add unit tests for activity_service — core CRUD
## Target Files
`tests/unit/test_activity_service.py`

## Description
Add `tests/unit/test_activity_service.py` covering core ActivityService methods: test_create_activity_returns_api_response, test_get_activity_found, test_get_activity_not_found, test_update_activity, test_delete_activity, test_list_activities_pagination.

## Acceptance
pytest tests/unit/test_activity_service.py -v — all 6 tests pass; ActivityService uses in-memory storage, no DATABASE_URL needed; fixture "activity_service" creates fresh instance per test

## Research Status
`plan.status: pass` — plan ready.

**Plan (6 tests):**
- `test_create_activity_returns_api_response`: assert bool(result) is True, result.data.id is not None, result.data.content matches
- `test_get_activity_found`: create then get, assert name matches
- `test_get_activity_not_found`: get id=9999, assert bool is False
- `test_update_activity`: update content, assert changed
- `test_delete_activity`: delete then get, assert not found
- `test_list_activities_pagination`: create 3, list page_size=2, assert len==2, total==3

**Risks:** activity_type string must match ActivityType enum values ("call", "email", "meeting", "note"); list_activities returns PaginatedData with .items as list of Activity objects (not dicts)

### PR DESCRIPTION
**Pull request by yihua-zhuo**
Closes #199

Plan: `.plans/issue-199.md`

This PR was auto-implemented by Claude Code in three stages:

1. **Plan** — Claude wrote a detailed implementation plan to `.plans/issue-199.md`.
2. **Implement** — Claude executed the plan, ran `ruff check src/` + the unit and integration test suites, and iterated until all three were green.
3. **Self-review** — Claude reviewed its own diff against `codereview.md` (the project review checklist) and fixed any rule violations found, re-running the test suite if any changes were made.

PR-time CI (`Dev Agent CI`) re-verifies independently before this PR is mergeable.

Please review before merging.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit test suite for activity service covering CRUD operations including create, retrieve, update, and delete functionality, plus list operations with pagination support. Tests validate data integrity, proper error handling for validation errors and missing records, and ensure overall reliability and robustness of core service features.

[Review Change Stack](https://app.coderabbit.ai/change-stack/yihua-zhuo/agent-job/pull/446)

### CODE REVIEW

**Comment by yihua-zhuo** [comment]
## Claude Code Review (inline)

**Status:** `pass`
**Inline comments posted:** 4

**Summary:** Reviewed 2 file(s) between origin/master and origin/auto/issue-199.

### Issues without a usable file/line
- `tests/unit/test_activity_service.py`: No test covers get_activity raising NotFoundException for a valid-id-but-deleted record (the activity exists then is deleted within the same test), which would catch edge cases in _fetch timing.

File: .plans/issue-199.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+## Goal
+Add `tests/unit/test_activity_service.py` with 6 unit tests covering the core CRUD methods of `ActivityService`. Tests run against in-memory mocks — no `DATABASE_URL` required.
+
+## Affected Files
+- `tests/unit/test_activity_service.py` — **new file**, contains all 6 tests
+- `src/services/activity_service.py` — read-only (implementation already exists)
+
+## Implementation Steps
+
+1. **Create `tests/unit/test_activity_service.py`** with a `MockActivityModel` class mirroring `ActivityModel` (id, tenant_id, customer_id, opportunity_id, type, content, created_by, created_at, to_dict) and a `MockResult` helper that returns rows via `scalar_one_or_none` and `scalars().all()`.
```
Comment (yihua-zhuo) [inline .plans/issue-199.md:14]: **INFO** (suggestion)

**Issue:** Plan describes a 'MockActivityModel class' but no such class exists; the implementation correctly uses a make_activity_handler + MockRow + MockResult composition instead.

**Suggestion:** Update the plan's language to reflect the actual approach: 'a MockRow-based activity handler' rather than a standalone MockActivityModel class. Not a code issue — plan language only.

File: .plans/issue-199.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- `activity_service` fixture creates a fresh `ActivityService(mock_session)` per test (function-scoped)
+- All async test methods use `pytest.mark.asyncio` decorator
+- `create_activity` returns an `Activity` domain object (NOT `ApiResponse`)
+- `list_activities` returns `tuple[list[Activity], int]` — assert on `.items` list length and total count directly
+- `delete_activity` returns `{"id": int}` — assert on dict key/value, not on an `Activity` object
+- `get_activity` raises `NotFoundException("活动记录")` for unknown IDs — assert with `pytest.raises(NotFoundException)`
+- `create_activity` raises `ValidationException` for unknown `activity_type` strings — assert with `pytest.raises(ValidationException)`
+- `activity_type` string values in test calls must match `ActivityType` enum values exactly: `"call"`, `"email"`, `"meeting"`, `"note"`
+
+## Risks / Open Questions
```
Comment (yihua-zhuo) [inline .plans/issue-199.md:48]: **INFO** (suggestion)

**Issue:** Plan 'Risks / Open Questions' section does not document the delete_activity returning dict vs Activity pattern, even though the plan itself flags this deviation.

**Suggestion:** Move the 'delete_activity returns dict' note from the body into the Risks section for completeness.

File: tests/unit/test_activity_service.py [hunk 3]
```diff
@@ -7,29 +7,12 @@
...
-
-@pytest.fixture
-def activity_service(mock_db_session):
-    return ActivityService(mock_db_session)
+# ---------------------------------------------------------------------------
+# Test helpers
+# ---------------------------------------------------------------------------
 
 
 def add_activity(
```
Comment (yihua-zhuo) [inline tests/unit/test_activity_service.py:18]: **WARNING** (suggestion)

**Issue:** The `add_activity` helper duplicates activity-handler record shape (field keys, created_at handling) and is redundant with make_activity_handler; it works correctly only because it happens to mirror the same field names the handler uses.

**Suggestion:** Extract `add_activity` into tests/unit/domain_handlers/activities.py as `seed_activity(state, **kwargs)` so the handler and test helpers share one source of truth for the record shape.

File: tests/unit/test_activity_service.py [hunk 4]
```diff
@@ -169,17 +196,26 @@ class TestDeleteActivity:
...
-        assert 1 not in activity_state.activities
 
-        with pytest.raises(NotFoundException):
-            await activity_service.get_activity(activity_id=1, tenant_id=1)
+        # Subsequent get_activity raises NotFoundException
+        with pytest.raises(NotFoundException, match="活动记录"):
+            await svc.get_activity(activity_id=1, tenant_id=1)
 
     @pytest.mark.asyncio
     async def test_delete_activity_rejects_wrong_tenant(self, activity_service, activity_state):
```
Comment (yihua-zhuo) [inline tests/unit/test_activity_service.py:221]: **INFO** (suggestion)

**Issue:** test_delete_activity_rejects_wrong_tenant uses the activity_service fixture (whose session has an empty state) rather than seeding a record into a fresh session, making the test less readable than the sibling tests that create their own session.

**Suggestion:** Use the same pattern as test_delete_activity: create a local session and svc with the seeded activity_state, then call svc.delete_activity.
